nixos/modules/services/networking/ddns-updater.nix

```
@@ -0,0 +1,51 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.ddns-updater;
+  defaultEnv = {
+    LISTENING_ADDRESS = ":8000";
+  };
+in
+{
+  options.services.ddns-updater = {
+    enable = lib.mkEnableOption "Container to update DNS records periodically with WebUI for many DNS providers";
+
+    package = lib.mkPackageOption pkgs "ddns-updater" { };
+
+    environment = lib.mkOption {
+      type = lib.types.attrsOf lib.types.str;
+      description = "Environment variables to be set for the ddns-updater service. DATADIR is ignored to enable using systemd DynamicUser. For full list see https://github.com/qdm12/ddns-updater";
+      default = defaultEnv;
```

```suggestion
      default = { };
      defaultText = lib.literalExpression ''
        {
          LISTENING_ADDRESS = ":8000";
        }
      '';
```

Default already set down below.